### PR TITLE
more image binding tests

### PR DESCRIPTION
**Pull request by baude**
add two additional bindings tests for image usage.

add ability to use search filter on the endpoint.

Signed-off-by: Brent Baude <bbaude@redhat.com>

### CODE REVIEW

**Comment by rhatdan** [comment]
You did a lot more then your comment indicates.  IE Fixed filtering.  Could you update your comment to indicate what was fixed.

**Comment by baude** [comment]
done

**Comment by rhatdan** [comment]
LGTM

File: pkg/api/handlers/images.go [hunk 1]
```diff
@@ -147,10 +148,36 @@ func SearchImages(w http.ResponseWriter, r *http.Request) {
...
+			if err != nil {
+				utils.InternalServerError(w, err)
+				return
+			}
+			filter.IsAutomated = types.NewOptionalBool(isAutomated)
+		}
+	}
 	options := image.SearchOptions{
-		Filter: image.SearchFilter{},
+		Filter: filter,
```
Comment (TomSweeneyRedHat) [inline pkg/api/handlers/images.go]: is there doc somewhere for "stars", "is-official" and "is-automated"?  or is this not customer facing?

Comment (baude) [reply pkg/api/handlers/images.go]: the api docs

**Pull request merged**